course-mw/enrolment-actor/src/main/scala/org/sunbird/enrolments/ContentConsumptionActor.scala

```diff
@@ -27,6 +27,20 @@ class ContentConsumptionActor @Inject() extends BaseEnrolmentActor {
     private val consumptionDBInfo = Util.dbInfoMap.get(JsonKey.LEARNER_CONTENT_DB)
     private val assessmentAggregatorDBInfo = Util.dbInfoMap.get(JsonKey.ASSESSMENT_AGGREGATOR_DB)
     val dateFormatter = ProjectUtil.getDateFormatter
+    val defaultFields = new java.util.ArrayList[String](){{
+        add("contentid")
+        add("userid")
+        add("batchid")
+        add("courseid")
+        add("completedcount")
+        add("completionpercentage")
+        add("lastcompletedtime")
+        add("progress")
+        add("status")
+        add("viewcount")
+    }}
+    val progressDetails = "progressDetails" //this is related to the response/request key
+    val progressdetails = "progressdetails" //this maps to DB column
```

Comment:
@pritha-tarento - do not hardcode JSON parsing only for progress details. Please create a list property named as "jsonFields" and convert to & from json for all the fields in this list

Reply:
Apply changes to generalized